<a href="https://colab.research.google.com/github/TaranSchlich/GB885---Assignment-7---Schlichtmann---T/blob/main/ops_report.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Bramble Bean — Operations Waste-Rate Report

Computes the company-wide monthly coffee **waste rate** across Bramble Bean's 40 cafés.

This notebook was also the subject of a version-control investigation: a prior commit had silently changed the waste-rate denominator from pounds **roasted** to pounds **sold**, understating the metric. The calculation below has been restored to the correct definition — waste divided by pounds roasted. See the repository README for the full investigation and fix.

## Load the café logs

Read the monthly per-café log (roasted, sold, and waste pounds) straight from a public raw GitHub URL, so the notebook reproduces anywhere with no manual upload.

In [ ]:
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/katiegaertner/bramble-ops-report/refs/heads/main/cafe_log.csv')

## Define the waste-rate metric

Waste is generated against what a café **roasts**, so the rate is `waste_lbs ÷ roasted_lbs`. (The reporting bug had used `sold_lbs` in the denominator.)

In [ ]:
def waste_rate(data):
    # tidied variable names
    total_waste_lbs = data['waste_lbs'].sum()
    total_roasted_lbs = data['roasted_lbs'].sum()
    return total_waste_lbs / total_roasted_lbs

## March waste rate

In [ ]:
march_df = df[df['month'] == 'March']
print(f"March Waste Rate: {waste_rate(march_df):.1%}")

March Waste Rate: 4.1%


In [ ]:
march_cafes = march_df['cafe_id'].nunique()
print(f"March summary: {waste_rate(march_df):.1%} waste across {march_cafes} cafes.")

March summary: 4.1% waste across 40 cafes.


## April waste rate

In [ ]:
april_df = df[df['month'] == 'April']
print(f"April Waste Rate: {waste_rate(april_df):.1%}")

April Waste Rate: 4.2%


In [ ]:
april_cafes = april_df['cafe_id'].nunique()
print(f"April summary: {waste_rate(april_df):.1%} waste across {april_cafes} cafes.")

April summary: 4.2% waste across 40 cafes.


## Report chart palette

Bramble Bean brand colors, kept on hand for styling report charts.

In [ ]:
# Bramble brand palette for report charts
BRAND_COLORS = {'espresso': '#4B3621', 'crema': '#D8C3A5', 'leaf': '#5B7553'}

## Findings

- **March:** 4.1% company-wide waste across 40 cafés.
- **April:** 4.2% company-wide waste across 40 cafés.

April's waste rate is slightly **higher** than March's. The earlier report that showed April *lower* (3.6%) was an artifact of the broken `÷ sold_lbs` calculation, not a real operational improvement.